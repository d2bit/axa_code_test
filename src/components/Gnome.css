.frame {
  background-color: #f9f9f9;
  margin: .3rem auto;
  padding: 1rem 2rem;
  border: 1px solid #ececec;
  flex-basis: 95%;
  display: flex;
  flex-flow: row wrap;
}

.title {
  margin: .2rem 0 0;
  font-size: 1.6rem;
}

.thumbnail {
  width: 100%;
  max-height: 200px;
  height: auto;
  border: 1px solid #b5b5b5;
}

.actions {
  flex-basis: 100%;
  margin-top: 1rem;
  display: flex;
  justify-content: space-around;
  align-self: flex-end;
}

.counter {
  margin: 0;
  border-bottom: 1px solid #c5c5c5;
  display: flex;
  flex-flow: column wrap;
  align-items: center;
}
.counterTitle {
  margin: 0;
  font-size: 1rem;
  font-weight: 600;
  text-transform: capitalize;
}
.counterValue {
  margin: 0;
  font-size: 3rem;
}

.btn {
  background-color: #4CAF50;
  font-size: 1.5rem;
  color: #ffffff;
  border: 1px solid #316934;
  border-radius: 50%;
}

@media (min-width: 550px) {
  .frame {
    flex-basis: 50%;
  }
}

@media (min-width: 850px) {
  .frame {
    flex-basis: 30%;
  }
}
