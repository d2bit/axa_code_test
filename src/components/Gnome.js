import React from 'react'
import style from './Gnome.css'

const Counter = ({ title, count }) => {
  return (
    <div className={style.counter}>
      <h1 className={style.counterTitle}>{title}</h1>
      <span className={style.counterValue}>{count}</span>
    </div>
  )
}

class Gnome extends React.PureComponent {
  render() {
    const { id, name, thumbnail, age, weitght, height, hair_color, professions, friends, openDetailsFn, ...rest } = this.props
    //console.log('creating gnome', this.props.id, this.props.name)
    //return <div>{id} - {name}</div>
    const openFn = () => openDetailsFn(id)
    return (
      <div className={style.frame} data-testid={`Gnome-${id}`} {...rest}>
        <h1 className={style.title} data-testid='name'>{name}</h1>
        <img className={style.thumbnail} width='200px' data-testid='thumbnail' src={thumbnail} alt={name} />
        <div className={style.actions}>
          <Counter title='Friends #' count={friends.length} />
          <Counter title='Professions #' count={professions.length} />
          <button className={style.btn} onClick={openFn}>OPEN</button>
        </div>
      </div>
    )
  }
}

export default Gnome
