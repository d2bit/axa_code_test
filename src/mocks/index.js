import data from './data.json'

const population = Object.values(data)[0]
const populationCount = population.length

export function getPopulation(size=5) {
  if (size >= populationCount) {
    return population
  }
  return population.slice(0, size)
}

export function getGnome(index) {
  if (index && index >= 0 && index < populationCount) {
    return population[index]
  }
  const randomIndex = Math.floor(Math.random() * populationCount)
  return population[randomIndex]
}

export function getResponseData() {
  return data
}

const Helper = {
  getPopulation,
  getGnome,
  getResponseData,
}

export default Helper
