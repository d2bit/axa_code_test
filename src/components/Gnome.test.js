import React from 'react'
import { render } from 'react-testing-library'
import 'dom-testing-library/extend-expect'
import { getGnome } from '../mocks'
import Gnome from './Gnome'

describe('Gnome', () => {
  it('shows the gnome information', () => {
    const gnomeId = 13
    const gnomeProfile = getGnome(gnomeId)

    const { getByTestId, container } = render(<Gnome {...gnomeProfile} />)

    expect(getByTestId('name')).toHaveTextContent(gnomeProfile.name)
    expect(getByTestId('thumbnail')).toHaveAttribute('src', gnomeProfile.thumbnail)
    expect(container).toMatchSnapshot()
  })
})
