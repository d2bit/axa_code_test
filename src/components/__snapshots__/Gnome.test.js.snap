// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gnome shows the gnome information 1`] = `
<div>
  <div
    data-testid="Gnome-13"
  >
    <h3
      data-testid="name"
    >
      Tobus Nozzletink
    </h3>
    <img
      alt="Tobus Nozzletink"
      data-testid="thumbnail"
      src="http://www.publicdomainpictures.net/pictures/120000/velka/cactus-1430656432Iug.jpg"
      width="200px"
    />
  </div>
</div>
`;
